packages/textarea/src/LionTextarea.js

```diff
@@ -59,17 +59,18 @@ export class LionTextarea extends ObserverMixin(LionInput) {
     this.maxRows = 6;
   }
 
-  connectedCallback() {
-    // eslint-disable-next-line wc/guard-super-call
-    super.connectedCallback();
-    this.setTextareaMaxHeight();
-    autosize(this.inputElement);
-  }
-
   disconnectedCallback() {
     autosize.destroy(this.inputElement);
   }
 
+  async firstUpdated() {
```

Comment:
that makes a lot of sense to finally have the initialisation in `firstUpdated`, I really don't understand how it worked before with just `connectedCallback` anywhere, not to say about Edge/IE11

what I find interesting here is that we now have `async firstUpdated()`, or in other words if someone needs to extend `LionTextarea`, they might think they can `await super.firstUpdated()` to be sure it was initialised

I think this is not a typical usage of this LitElement life-cycle hook, so it is better to not make this method async and just use a promise inside (but do not return it), so that public API does not provide a possibility to do `await super.firstUpdated()`

if someone will ever need such a hook, we can always add it later, but it should not be in `firstUpdated`, maybe it will be a modified `updateComplete`, I mean smth that is expected to be async by the user of LitElement

Reply:
I don't clearly understand either, I didn't dig deep enough on the webcomponents polyfill, but it's probably related to what is done on the LitElement's `connectedCallback`, I think it has a slightly different rendering behavior when the polyfill is needed because of the ShadyCSS:
```
connectedCallback() {
    super.connectedCallback();
    // Note, first update/render handles styleElement so we only call this if
    // connected after first update.
    if (this.hasUpdated && window.ShadyCSS !== undefined) {
      window.ShadyCSS.styleElement(this);
    }
}
```
Maybe I'll try to ask Justin Fagnani or the polymer team on twitter about the subject to see if they can clear it out for us, or/and do some tests and maybe open an issue to them, but don't worry, I'll not talk about the project since it's asked not to on the readme ;)

You're totally right, it's going to be bad to inherit later, I'll update my commit to keep it without the async and use the `then` to synchronize with the `updateComplete` ;D